controllers/activities.js

```diff
@@ -21,6 +23,7 @@ function createAddNoteActivity({ previousValue, newValue, createdAt, twilioWorke
     type: ActivityTypes.addNote,
     text: newNote,
     twilioWorkerId,
+    icon: ActivityTypes.addNote,
```

Comment:
I am very skeptical about this.  Storing which display icon to use in our backend feels like we're storing presentation information in a place it shouldn't be.  This is unnecessary coupling of two separate systems.  I'm also confused why this is needed since it isn't used in https://github.com/tech-matters/flex-plugins/pull/263/files. Why do this?

Reply:
Originally, this was called `channel` indicating that it's the actual channel that the contact came from (when it's a contact type activity with `default` channel, we want the information of the counselor-input-channel, like facebook, whatsapp, etc), but changed to icon just to be consistent with note-type activities. I could make this to be `channel` and  notes will have that property undefined. Does that makes more sense?

Reply:
I'm sorry, I actually misunderstood what this was doing at first.  I was hurrying to review in hopes of finishing before the end of your day.  I thought that we were saving an `icon` value into Postgres.

Reading through more of the code, I understand better.  We don't store activities in the database, we just produce them from the `CaseAudits` table and connected contacts.

I still think sending icon is strange, as it's very presentation specific.  I don't necessarily think that the same fields have to be sent for a note vs a contact.  I think there are two options:

1. Decide that the client doesn't need to know or care if the `type` we return comes from the `channelType` or contactlessTask.channel`.  Return whatever `type` we want them to show.  We could possibly add another field: `offlineContact: true|false` (or `isContactless` if you prefer).  So something like:

```
return   return {
    date: createdAt,
    type: type === ActivityTypes.connectContact.default
      ? newContact.rawJson.contactlessTask.channel
      : type
    text: newContact.rawJson.caseInformation.callSummary,
    twilioWorkerId,
    offlineContact: type === ActivityTypes.connectContact.default  // not sure this is needed
  };
```

2. Use the same structure as with `icon` but name it differently.  Maybe `apparentType`.

I lean toward #1 because I just don't think the client needs to care whether it was an offline contact.  I think the goal of adding offline contacts is that once it's added, the system believes that it happened when the user said it did.  We still want an audit history to be able to check, but otherwise it shouldn't matter.

Reply:
Ok, so I've tried doing #1 as I like that one the most, but the fact that there may be "channels" that are not actually part of our communication channels, made impossible the check in the frontend weather the timeline activity should be treated as a contact or not. So I'm sending the `channel` that the counselor inputs as a separate property of the activity object, so, `type` being `default`indicates this is a contact, and for the icon we use `channel` extra property. Let me know if this makes sense to you.